cpp/src/gandiva/decimal_ir.cc

```diff
@@ -353,6 +353,60 @@ Status DecimalIR::BuildSubtract() {
   return Status::OK();
 }
 
+Status DecimalIR::BuildMultiply() {
+  // Create fn prototype :
+  // int128_t
+  // multiply_decimal128_decimal128(int128_t x_value, int32_t x_precision, int32_t
+  // x_scale,
+  //                           int128_t y_value, int32_t y_precision, int32_t y_scale
+  //                           int32_t out_precision, int32_t out_scale)
+  auto i32 = types()->i32_type();
+  auto i128 = types()->i128_type();
+  auto function = BuildFunction("multiply_decimal128_decimal128", i128,
+                                {
+                                    {"x_value", i128},
+                                    {"x_precision", i32},
+                                    {"x_scale", i32},
+                                    {"y_value", i128},
+                                    {"y_precision", i32},
+                                    {"y_scale", i32},
+                                    {"out_precision", i32},
+                                    {"out_scale", i32},
+                                });
+
+  auto arg_iter = function->arg_begin();
+  ValueFull x(&arg_iter[0], &arg_iter[1], &arg_iter[2]);
+  ValueFull y(&arg_iter[3], &arg_iter[4], &arg_iter[5]);
+  ValueFull out(nullptr, &arg_iter[6], &arg_iter[7]);
+
+  auto entry = llvm::BasicBlock::Create(*context(), "entry", function);
+  ir_builder()->SetInsertPoint(entry);
+
+  // Make call to pre-compiled IR function.
+  // TODO: add fast paths.
```

Comment:
Not sure I understand this comment. There is already a fast path, no? The only issue might be that `gandiva::decimalops::Multiply` isn't inlined into its caller.

Reply:
I wanted to try the intrinsics-with-overflow instructions. I've removed the comment - will experiment with this after adding a micro benchmark.